esiclient/v1/orchestrator/openshift.py

```diff
@@ -0,0 +1,368 @@
+        }
+
+        print("STARTING OPENSHIFT CLUSTER INSTALL")
+
+        # get cluster id
+        if not cluster_id:
+            try:
+                cluster_data = {
+                    "name": cluster_name,
+                    "openshift_version": openshift_version,
+                    "high_availability_mode": high_availability_mode,
+                    "base_dns_domain": base_dns_domain,
+                    "api_vip": api_vip,
+                    "ingress_vip": ingress_vip,
+                    "ssh_public_key": ssh_public_key,
+                    "pull_secret": pull_secret
+                }
+                response = call_assisted_installer_api(
+                    'clusters', 'post', headers, cluster_data)
+                cluster_id = response.json().get('id')
+            except Exception as e:
+                self._print_failure_message(
+                    e,
+                    parsed_args.cluster_config_file,
+                    message='Error creating OpenShift cluster')
+                raise OrchestrationException()
+        print("cluster ID: %s" % cluster_id)
+
+        # get infra env id
+        if not infra_env_id:
+            try:
+                infra_env_data = {
+                    "name": "%s-infra-env" % cluster_name,
+                    "image_type": "minimal-iso",
+                    "cluster_id": cluster_id,
+                    "pull_secret": pull_secret,
+                    "openshift_version": openshift_version,
+                    "ssh_authorized_key": ssh_public_key
+                }
+                response = call_assisted_installer_api(
+                    'infra-envs', 'post', headers, infra_env_data)
+                infra_env_id = response.json().get('id')
+            except Exception as e:
+                self._print_failure_message(
+                    e,
+                    parsed_args.cluster_config_file,
+                    cluster_id=cluster_id,
+                    message='Error creating OpenShift cluster infra env')
+                raise OrchestrationException()
+        print("infra env ID: %s" % infra_env_id)
+
+        # register nodes
+        try:
+            response = call_assisted_installer_api(
+                "infra-envs/%s/hosts" % infra_env_id,
+                'get', headers)
+            host_count = len(response.json())
+            if host_count < len(nodes):
+                response = call_assisted_installer_api(
+                    "infra-envs/%s/downloads/image-url" % infra_env_id,
+                    'get', headers)
+                image_url = response.json().get('url')
+                provisioning_network = neutron_client.find_network(
+                    provisioning_network_name)
+                print("provisioning nodes")
+                for node_name in nodes:
+                    node = ironic_client.node.get(node_name)
+                    if node.provision_state == 'available':
```

Comment:
Do we want to make sure that all nodes are in 'available' state before registering nodes?